Remotion/Data/DomainObjects/Infrastructure/ObjectIDConverter.cs

```diff
@@ -0,0 +1,83 @@
+  /// <param name="context"> An <see cref="ITypeDescriptorContext"/> that provides a format context. </param>
+  /// <param name="destinationType"> The <see cref="Type"/> to convert an <see cref="ObjectID"/> value to. </param>
+  /// <returns> <see langword="true"/> if the conversion is supported. </returns>
+  public override bool CanConvertTo (ITypeDescriptorContext? context, Type? destinationType)
+  {
+    if (destinationType == null)
+      return false;
+
+    return destinationType == typeof(string) || base.CanConvertTo(context, destinationType);
+  }
+
+  /// <summary> Converts <paramref name="value"/> into an <see cref="ObjectID"/> value. </summary>
+  /// <param name="context"> An <see cref="ITypeDescriptorContext"/> that provides a format context. </param>
+  /// <param name="culture"> The <see cref="CultureInfo"/> to use as the current culture. </param>
+  /// <param name="value"> The source value. </param>
+  /// <returns> An <see cref="ObjectID"/> value.  </returns>
+  /// <exception cref="NotSupportedException"> The conversion could not be performed. </exception>
+  public override object? ConvertFrom (ITypeDescriptorContext? context, CultureInfo? culture, object? value)
+  {
+    if (value == null)
+      return null;
+
+    if (!CanConvertFrom(context, value.GetType()))
+      throw new NotSupportedException($"Value of type '{value.GetType()}' cannot be converted to type 'ObjectID'.");
+
+    if (value is string valueString)
+    {
+      if (ObjectID.TryParse(valueString, out var objectID))
+        return objectID;
+      else
+        throw new NotSupportedException($"String '{valueString}' cannot be converted to type 'ObjectID'.");
+    }
+
+    return base.ConvertFrom(context, culture, value);
+  }
+
+  /// <summary> Converts an <see cref="ObjectID"/> value into the <paramref name="destinationType"/>. </summary>
+  /// <param name="context"> An <see cref="ITypeDescriptorContext"/> that provides a format context. </param>
+  /// <param name="culture"> The <see cref="CultureInfo"/> to use as the current culture. </param>
+  /// <param name="value"> The <see cref="ObjectID"/> value to be converted. </param>
+  /// <param name="destinationType"> The destination <see cref="Type"/>. Must not be <see langword="null"/>. </param>
+  /// <returns> An <see cref="Object"/> that represents the converted <see cref="ObjectID"/> converted to a <see cref="String"/>.</returns>
+  /// <exception cref="NotSupportedException"> The conversion could not be performed. </exception>
+  public override object? ConvertTo (ITypeDescriptorContext? context, CultureInfo? culture, object? value, Type destinationType)
+  {
+    ArgumentUtility.CheckNotNull(nameof(destinationType), destinationType);
+
+    if (value == null)
+      return null;
+
+    if (!CanConvertTo(destinationType))
+      throw new NotSupportedException($"This TypeConverter cannot convert ObjectIDs to type '{destinationType}'.");
+
```

Comment:
Same as above, not needed as we have the base call.

Reply:
I'm not sure that accurate. Looking at 
```
public virtual object ConvertTo(
      ITypeDescriptorContext context,
      CultureInfo culture,
      object value,
      Type destinationType)
    {
      if (destinationType == (Type) null)
        throw new ArgumentNullException(nameof (destinationType));
      if (!(destinationType == typeof (string)))
        throw this.GetConvertToException(value, destinationType);
      if (value == null)
        return (object) string.Empty;
      return culture != null && culture != CultureInfo.CurrentCulture && value is IFormattable formattable ? (object) formattable.ToString((string) null, (IFormatProvider) culture) : (object) value.ToString();
    }
```
And there's an exception for anything but string.

Reply:
Thats weird then, I tried it and got no exception in the test, but maybe I messed up the test setup somehow.

Reply:
Yea, worked now, I probably messed up the test somehow